docs/readmes/orc8r/hold_terraform_state_on_s3.md

```diff
@@ -0,0 +1,100 @@
+---
+id: hold_terraform_state_on_s3
+title: Hold Terraform state outside my local machine
+hide_title: true
+---
+# Holding the terraform state outside of my local machine
+
+**Description:** It is best to not hold the terraform state file locally on any disk that is susceptible to local failures and loss of state file. Note that the state file is the blueprint of the assets deployed on the public cloud environment and if this gets corrupt, it will be very difficult to manage cloud assets in a programmatic manner.
+
+To aid this vision, terraform supports holding the state file remotely. One example is to use the AWS S3 bucket to store the state file. However, in such a scenario, it is plausible that more than one user may try to access the state file at the same time. For this reason, we can use a dynamo database to lock the database when it is being read and written to
+
+**Environment:** Orc8r in AWS/EKS
+
+**Components:** Orchestrator
+
+**Steps:**
+
+High level steps:
+
+1. [Create an S3 bucket](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/create-bucket.html)
+
+2. [Create a dynamo dB state locking](https://www.terraform.io/docs/language/settings/backends/s3.html#dynamodb-state-locking)
+
+3. Add configuration to the main.tf file to change the backend(variables `state_backend` and `state_config`)
+
+```
+module orc8r-app {
+    source = "github.com/magma/magma//orc8r/cloud/deploy/terraform/orc8r-helm-aws?ref=v1.3.2"
+
+    region = "us-west-2"
+
+    orc8r_domain_name = module.orc8r.orc8r_domain_name
+    orc8r_route53_zone_id = module.orc8r.route53_zone_id
+    external_dns_role_arn = module.orc8r.external_dns_role_arn
+
+    secretsmanager_orc8r_name = module.orc8r.secretsmanager_secret_name
+    seed_certs_dir = "/Users/$USER/secrets/certs"
+
+    orc8r_db_host = module.orc8r.orc8r_db_host
+    orc8r_db_name = module.orc8r.orc8r_db_name
+    orc8r_db_user = module.orc8r.orc8r_db_user
+    orc8r_db_pass = module.orc8r.orc8r_db_pass
+
+    nms_db_host = module.orc8r.nms_db_host
+    nms_db_name = module.orc8r.nms_db_name
+    nms_db_user = module.orc8r.nms_db_user
+    nms_db_pass = module.orc8r.nms_db_pass
+
+    docker_registry = "something-orc8r-docker.registry.com"
+    docker_user = "$UPERU$ER"
+    docker_pass = "$UPER$3CRETPA$$W0RD"
+
+    helm_repo = "https://some-helm-repo"
+    helm_user = "$UPERU$ER"
+    helm_pass = "$UPER$3CRETPA$$W0RD"
+
+    eks_cluster_id = module.orc8r.eks_cluster_id
+
+    efs_file_system_id = module.orc8r.efs_file_system_id
+    efs_provisioner_role_arn = module.orc8r.efs_provisioner_role_arn
+
+    elasticsearch_endpoint = module.orc8r.es_endpoint
+
+    orc8r_chart_version = "1.4.36"
+    orc8r_tag = "v1.3.2"
+    deploy_nms = true
+    state_backend = "s3"
+    state_config = {
+        bucket = "s3_bucket_name"
+        region = "us-west-2"
```

Comment:
I think it would be good to just link to one of our template terraform files and only include the relevant config here

Reply:
Otherwise it may be tricky to keep this in sync with the template terraform file

Reply:
thanks for the feedback @andreilee ! comments have been addressed.